### ISSUE TITLE
Use file name as instance name in all templates

### ISSUE THREAD

**Issue by rcdailey** [recyclarr/config-templates#19]
I think it would be a good idea to have a unique instance name (by default) in every configuration file. Specifically, I think the instance name should follow the file name. This has a few benefits:

- It's easier to identify the instances you can run by looking at the file names.
- It's one less manual edit users have to make.
- If users want to merge instances into a single file, that becomes easier too.

As an example, if we look at `radarr/uhd-bluray-web.yml`:

```yml
# Updated: 2023-07-19
radarr:
  radarr:  # A custom name used to identify this particular instance.
    base_url: Put your Radarr URL here
    api_key: Put your API key here
```

I propose that we rename the instance (for this one file) to `uhd-bluray-web`, like so:

```yml
# Updated: 2023-07-19
radarr:
  uhd-bluray-web:  # A custom name used to identify this particular instance.
    base_url: Put your Radarr URL here
    api_key: Put your API key here
```

How does this sound to you @yammes08?

**Comment by yammes08** [comment]
A couple of thoughts:

- If people are merging multiple configs into one file for a single instance, and each config has a different instance name, won't users get errors relating to CF ownership as each copied in config will be for a different 'instance'?
- Users who use a single config, and currently just copy/paste in updates (or run some sort of diff script type thing), will they also start to get CF ownership errors if the instance name is different?

I'm totally open to changing the instance names in the templates, in fact it's probably the thing I've put the least thought into. I just want to see if we can bottom out what the impact might be on folks already using them.

**Comment by rcdailey** [comment]
> * If people are merging multiple configs into one file for a single instance, and each config has a different instance name, won't users get errors relating to CF ownership as each copied in config will be for a different 'instance'?
> * Users who use a single config, and currently just copy/paste in updates (or run some sort of diff script type thing), will they also start to get CF ownership errors if the instance name is different?

Both of these points seem to describe the same scenario. Can you perhaps clarify the difference?

If people are manually merging multiple templates into a single file for the purpose of syncing everything to one instance, the approach will be different. Even if we do not change instance names at all, they should not be copying the `radarr:` part. I recently [added a config example](https://recyclarr.dev/wiki/yaml/config-examples/#merge-single-instance) for this scenario and it walks through the steps users would need to take.

Basically, I don't expect the instance name in the templates to play any part in this.

> I'm totally open to changing the instance names in the templates, in fact it's probably the thing I've put the least thought into. I just want to see if we can bottom out what the impact might be on folks already using them.

I think that's a fantastic idea and in fact is the reason I opened this issue. Let's keep working through your concerns and see if we can get on the same page.

I appreciate your reply!

**Comment by yammes08** [comment]
> Both of these points seem to describe the same scenario. Can you perhaps clarify the difference?

I think point one was for new users who are merging multiple configs into a single config for the first time, but your linked wiki page (which I hadn't seen!) explains this to users perfectly, so we can rule that out.

Point two was for existing users who might just be copy/pasting the whole config every time there's an update (then amending their base_url and api_key, or secrets, or running some sort of diff to bring in new/updated lines. Those folks could potentially encounter an error when they're running an updated version of the file but the instance name has changed.

**Comment by rcdailey** [comment]
> Those folks could potentially encounter an error when they're running an updated version of the file but the instance name has changed

I suppose this would be possible if they explicitly referenced instance names in a script that runs sync commands or something. But these users would already have edited the instance name by hand because `radarr` is used in more than one template right now, and it's not valid to have two instances of the same name. So they'd already be aware of the need to skip these. If anything, this gives them the opportunity to eliminate 1 extra line diff from their comparison when they pull in updates.

**Comment by yammes08** [comment]
> I suppose this would be possible if they explicitly referenced instance names in a script that runs sync commands or something. But these users would already have edited the instance name by hand because `radarr` is used in more than one template right now, and it's not valid to have two instances of the same name. So they'd already be aware of the need to skip these. If anything, this gives them the opportunity to eliminate 1 extra line diff from their comparison when they pull in updates.

Cool, makes sense. Happy to make this change next time I do an update (waiting for some pending guide changes so I can do it all at the same time).
